tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5525,21 +5525,20 @@ void SpirvEmitter::storeValue(SpirvInstruction *lhsPtr,
     // parameters/returns/variables of type T. And ConstantBuffer<T> is not
     // represented differently as struct T.
   } else if (isOpaqueArrayType(lhsValType)) {
+    // SPIRV-Tools can handle legalization of the store in these cases.
+    if (!lhsValType->isConstantArrayType() || rhsVal->isRValue()) {
+      spvBuilder.createStore(lhsPtr, rhsVal, loc);
+      needsLegalization = true;
+      return;
+    }
+
     // For opaque array types, we cannot perform OpLoad on the whole array and
```

Comment:
@jaebaek I know we talked about removing this comment as well, but as far as I can tell it looks to still be correct, but it applied to the later block, so I moved it down for clarity. Let me know if that still doesn't seem right!

Reply:
Makes sense. Thanks.